src/sumBy.ts

```diff
@@ -40,18 +58,34 @@ export function sumBy<T>(
   callbackfn: (value: T, index: number, data: ReadonlyArray<T>) => number,
 ): number;
 
+export function sumBy<T>(
+  data: ReadonlyArray<T>,
+  callbackfn: (value: T, index: number, data: ReadonlyArray<T>) => bigint,
+): bigint | 0;
+
 export function sumBy(...args: ReadonlyArray<unknown>): unknown {
   return purry(sumByImplementation, args);
 }
 
 const sumByImplementation = <T>(
   array: ReadonlyArray<T>,
-  callbackfn: (value: T, index: number, data: ReadonlyArray<T>) => number,
-): number => {
-  let sum = 0;
+  callbackfn: (
+    value: T,
+    index: number,
+    data: ReadonlyArray<T>,
+  ) => bigint | number,
+): bigint | number => {
+  let sum: bigint | number | undefined;
   for (const [index, item] of array.entries()) {
     const summand = callbackfn(item, index, array);
-    sum += summand;
+    if (sum === undefined) {
+      sum = summand;
+      continue;
+    }
+    // we use 'as number' to ensure TypeScript treats
+    // 'sum' and 'summand' as the same type,
+    // which could be a bigint.
+    (sum as number) += summand as number;
```

Comment:
**suggestion:** let's use an explicit `ts-expect-error` rather than a typecast; see the similar thing in `sum`

Reply:
thx for your suggestion. I have made the changes, so please review it again